text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+
+#### `custom-service`
+This plugin is meant to provide a replacement for `rtmp_custom` type.
+
+If OBS need to be heavily dependent to one plugin, it shall be this one.
+
+The protocol will be detected (or maybe forced set by the user).
+Maybe properties will show depending of the protocol.
+
+#### `obs-services`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who doesn't have custom behavior or integration.
+
+This plugin will need a brand new services.json with new format, to register each streaming service with their own id. No more things like `rtmp-common` id.
+ 
+Those services will be able to provide multiple protocols so no more "Service - HLS" and "Service - FTL".
+
+If a certain protocol is not available, the plugin will not register the service if the service doesn't use another protocol.
+If it does, the missing protocol will not be shown. [Thanks to RFC 45](https://github.com/tytan652/rfcs/blob/protocol_api/text/0045-protocols-in-outputs-api.md).
+
+The notion of maximum and recommended shall be distinguished.
+Maximum, supported resolutions, recommended settings will be ignorable with their respective checkboxes.
+
+Those services should have no specific behavior like ingest management.
+
+But can list with which codec they are compatible if needed. [Thanks to RFC 45](https://github.com/tytan652/rfcs/blob/protocol_api/text/0045-protocols-in-outputs-api.md)
+
+In the end, adding streaming service will only be a JSON addition for this plugin.
+Only improvements will be accepted in the code of this plugin.
+
+#### A plugin per special case
+This/Those plugin(s) is/are meant to provide replacements for `rtmp_common` type for services which need a custom behavior.
+
+This/Those plugin(s) is/are meant to have implemented and register services which need a specific behavior like custom ingest with a unique id for each one.
+
+#### Note about the plugins
+- Like this if the service needs a specific behavior, you can 'transfer' it from `obs-services` to a new plugin and keep the same id. The change will be seamless for the user.
+
+### Conversion of services
+**Downgrade will break service configuration**
+A JSON or a harcoded list with old service name linked to their new id, to make OBS able to convert the `service.json` to a new one.
+
+Auth integration config inside `basic.ini` will be also transfered to the new file.
+
+## Other changes
+
+### Service API changes
+- Add missing get_properties2 and get_default2 functions for service
+
+### Property API addition
+- Add a property to be able to show information with text label
+
+Since the service API is in a way unusable by third-party, any breakage will only impact OBS built-in service plugins ( so only `rtmp-services`).
+
+### Service integrations new flow
+
+#### How docks and CEF widget for OAuth will be created
+For the docks we could add the possibility to add them through the frontend API.
```

Comment:
Additionally, it should be able to add a dock without making it visible - this is how most of the Twitch docks work, we add them but don't display them to avoid overwhelming the user with too many new docks at once. Any API for managing these docks should be able to decide if said docks are docked by default and where.